library/packages/src/lib/y2packager/product_license.rb

```diff
@@ -0,0 +1,93 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2018 SUSE LLC, All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+# ------------------------------------------------------------------------------
+
+require "yast"
+require "forwardable"
+require "y2packager/license"
+require "y2packager/licenses_handlers"
+
+module Y2Packager
+  class ProductLicense
+    # This class holds the license stuff for a given product
+    #
+    # Why a separate ProductLicense class? First of all, we wanted to extract
+    # the license handling from Y2Packager::Product and moving this logic to
+    # Y2Packager::License was not a good idea because different products could
+    # share the same license. Additionally, this class offers an API to work
+    # with licenses with a proper Product or Addon object is not available
+    # (backward compatibility reasons).
+    #
+    # @see Y2Packager::Product
+    # @see Y2Packager::License
+
+    extend Forwardable
+
+    def_delegators :@license, :content_for, :locales, :accept!, :reject!
+
+    # @!method license_confirmation_required?
+    #   Determine whether the license should be accepted or not
+    #   @return [Boolean] true if the license acceptance is required
+    #
+    # @!method license_confirmation=(confirmed)
+    #   Set the license confirmation for the product
+    #   @param confirmed [Boolean] true if it should be accepted; false otherwise
+    def_delegators :@handler, :license_confirmation_required?, :license_confirmation
+
+    # @return [License] Product's license
+    attr_reader :license
+
+    class << self
+      # Find license for a given product
+      #
+      # @param product_name [String]    Product's name
+      # @param source       [:rpm,:url] Source to get the license from. For the time being,
+      #   only :rpm is really supported.
+      # @return [ProductLicense]
+      def find(product_name, source = :rpm, options = {})
```

Comment:
to be honest, always when I see two default params, I get to my mind that keyword params would be better.

Reply:
also this `options = {}` is everywhere and smell a bit for me. I think it was intended as possible future extension of functionality, but with keyword params, you can also easily add it and do not need to have holder.